src/Message/ImportCommandCreator.php

```diff
@@ -0,0 +1,18 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Message;
+
+use App\MessageHandler\BitbucketImportCommandHandler;
+use App\MessageHandler\GithubImportCommandHandler;
+
+class ImportCommandCreator {
+    public function create($username, $provider) {
+        if ($provider == 'github') {
+            return new GithubImportCommand($username);
+        }
+        if($provider == 'bitbucket') {
+            return new BitbucketImportCommand($username);
+        }

```

Comment:
There we can do exercise: How to implement each new provider without needed of if or switch here? As we use AbstractFactory it should be possible

Reply:
Fixed